.github/workflows/layer-publish.yml

```diff
@@ -47,6 +51,9 @@ jobs:
           
           if [[ -n "${{ inputs.architecture }}" ]]; then
             LAYER_NAME=$LAYER_NAME-${{ inputs.architecture }}
+            ARCH="${{ inputs.architecture }}"
+          else
+            ARCH="x86_64 arm64"
```

Comment:
do we have any layer that is ARCH agnostic, i.e.: supports both x86_64 and arm64? As far I'm aware all the layers include the collector, which is architecture dependent.

Reply:
The collector is an independent layer from the automatic instrumentation. So both layers must be declared. As a result, most (all?) auto instrumentation layers are arch independent.

Reply:
you are right. I was confused by the aws lambda layers. 

Still on the `ARCH` environment variable, I think you need to echo the env var into `$GITHUB_ENV` for it to be accessible in subsequent steps, similarly to how it is done to `LAYER_NAME`.

reference: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-environment-variable

Reply:
nit: what do you think about making the `architecture` github input obligatory and defining explicitly in each workflow that calls `layer-publish.yaml` what are the supported architectures? this would simplify a tiny bit this workflow by removing the need of the the intermediate `ARCH` env var.

Reply:
Good catch on `$GITHUB_ENV`.  I've updated that now.

As for making `architecture` required, I think it would be a tradeoff.  It would shift the complexity from here to the layer naming.  When we support both, we don't want it included in the name. I think it's easier to keep as is.